### PR TITLE
Add bash_bug_uncaught_unbound_on_exit_trap

### PR DESCRIPTION
**Pull request by andyneff**
Signed-off-by: Andy Neff <andy@visionsystemsinc.com>

### CODE REVIEW

File: linux/compat.bsh [hunk 1]
```diff
@@ -477,17 +477,26 @@ function load_vsi_compat()
   #     # In bash >= 4:
   #     # y=
   #     # 0
+  #
+  # .. var :: bash_bug_uncaught_unbound_on_exit_trap
+  #
+  # There is a bug in bash 3.2, where ``set -u`` will not return non-zero for an unbound variable if an ``EXIT`` trap is set. Unfortunately, there is no way to detect this happens in the ``EXIT`` trap code, nor is there any way to cope for it. This means an unbound variable could potentially go undetected.
```
Comment (scott-vsi) [inline linux/compat.bsh:483]: cope for it -> cope with it

Comment (scott-vsi) [reply linux/compat.bsh:483]: Is this a problem in bash 4.0 or 4.1?

File: linux/compat.bsh [hunk 2]
```diff
@@ -477,17 +477,26 @@ function load_vsi_compat()
...
+  #
+  # :Value: * ``0`` Bash has the bug
+  #         * ``1`` Unbound variable exit as expected
```
Comment (scott-vsi) [inline linux/compat.bsh:486]: exit -> exits

File: linux/compat.bsh (hunk 1 above)
Comment (andyneff) [reply linux/compat.bsh:483]: No, I checked, bash 3.2 only

**Review by scott-vsi — APPROVED** [review]

**Pull request merged**